src/scripts/train/sft/OLMo-sft.py

```diff
@@ -0,0 +1,629 @@
+            self.grad_accum_steps = 1
+
+        # Validate that rank_microbatch_size_tokens is divisible by sequence_length
+        assert self.rank_microbatch_size_tokens % self.sequence_length == 0, (
+            "rank_microbatch_size_tokens must be divisible by sequence_length (got "
+            f"{self.rank_microbatch_size_tokens} and {self.sequence_length})"
+        )
+        self.rank_microbatch_size_sequences = (
+            self.rank_microbatch_size_tokens // self.sequence_length
+        )
+
+        # Final validation
+        total_tokens = self.rank_microbatch_size_tokens * dp_world_size * self.grad_accum_steps
+        assert self.global_batch_size_tokens == total_tokens, (
+            "global_batch_size_tokens must equal "
+            "(rank_microbatch_size_tokens * dp_world_size * grad_accum_steps) (got "
+            f"{self.global_batch_size_tokens} and {total_tokens})"
+        )
+
+
+def _separate_prefix_and_glob(prefix: str) -> Tuple[str, str]:
+    if any(char in prefix for char in ["*", "?", "[", "]"]):
+        parts = prefix.split("/")
+        base_parts = []
+        for part in parts:
+            if any(char in part for char in ["*", "?", "[", "]"]):
+                break
+            base_parts.append(part)
+    else:
+        base_parts = prefix.split("/")
+    if not base_parts:
+        return ".", prefix
+
+    new_prefix = "/".join(base_parts)
+    glob_str = prefix[len(new_prefix) :]
+
+    return new_prefix, glob_str.lstrip("/")
+
+
+def glob_remote_dataset(prefix: str) -> List[str]:
+    parsed_path = urlparse(prefix)
+    scheme, bucket, parsed_prefix = (
+        parsed_path.scheme,
+        parsed_path.netloc,
+        parsed_path.path.lstrip("/"),
+    )
+    parsed_prefix_pre_glob, glob_str = _separate_prefix_and_glob(parsed_prefix)
+    base_prefix_without_scheme = Path(f"{bucket}/{parsed_prefix_pre_glob}")
+
+    paths: List[str] = []
+
+    for path in list_directory(f"{scheme}://{base_prefix_without_scheme}"):
+        parsed_path = urlparse(path)
+        path_without_scheme = Path(parsed_path.netloc) / parsed_path.path.lstrip("/")
+        relative_to_base_prefix = path_without_scheme.relative_to(base_prefix_without_scheme)
+
+        if glob_str and not fnmatch.fnmatch(str(relative_to_base_prefix), glob_str):
+            # must match the glob if a glob was provided
+            continue
+
+        # path was valid!
+        paths.append(path)
+
+    return paths
+
+
+def build_sft_dataset(
+    root_dir: str,
+    tokenizer_config: TokenizerConfig,
+    sequence_length: int,
+    dataset_path: Optional[str],
+) -> NumpyPackedFSLDatasetConfig:
+    clean_path = dataset_path.rstrip("/")
+    if dataset_path.startswith("gs://") or dataset_path.startswith("s3://"):
+        token_id_paths = glob_remote_dataset(f"{clean_path}/token_ids_part_*.npy")
+        label_mask_paths = glob_remote_dataset(f"{clean_path}/token_ids_part_*.npy")
```

Comment:
globing remote datasets should work out of the box now! no need for this if / else block and just keep `expand_glob=True`